composer/yamls/models/classify_mnist.yaml

```diff
@@ -41,5 +41,5 @@ dataloader:
   persistent_workers: true
   num_workers: 8
 validate_every_n_epochs: 1
-grad_accum: 1
+grad_accum: -1
```

Comment:
Can we actually just make grad_accum an `Optional[int]` type in the trainer and have it default to `None` (meaning do autotuning). 

That way we can remove it as a parameter from the YAML files entirely rather than having to specify -1 everywhere.

Reply:
Hm, right now it is an optional int defaulting to -1, so I can just get rid of the -1 in all the yamls. 

I'm personally stylistically in favor of -1 over None -- in general, None seems to specify to me don't do this whereas -1 is closer to saying it's not user specified. I'm fine with it either way though if there's a more consistent practice in the codebase

Reply:
Since this is an experimental feature, I'm wary of making it the default until we've had it bake in the trainer for a bit. So perhaps we should defualt still to `grad_accum = 1`. btw, grad_accum='auto' is also an often-used pattern.

Reply:
> Since this is an experimental feature, I'm wary of making it the default until we've had it bake in the trainer for a bit. So perhaps we should defualt still to `grad_accum = 1`. 

It sounds to me like we all would prefer it default since it's one less thing for a user to think about (if this isn't the case, happy to discuss and switch). In that case, it seems to me you're saying we shouldn't push this as default until we're confident it works. I'm happy to switch it non-default, but it'd be helpful to understand what criteria we need to hit to make it default. My sense is pushing it in dev now will mean we get more internal testing before any release, whereas if it isn't default, it's going to be pretty rare for people to switch it on and we won't actually be more confident with it as time passes. Still picking up best practices with composer development, so feel free to correct me / share more detail :) 

> btw, grad_accum='auto' is also an often-used pattern.

This seems like a cleaner pattern -- will switch unless anyone else has other ideas

Reply:
One way to test could be to make it `auto` in config yamls but by default 1 (as before) in the actual codebase, so we get a fair amount of internal testing still

Reply:
Yeah the feature is awesome, but want to let it bake for a bit to flush out any odd issues before changing the defaults. Here's what I suggest:

1. keeping the default on 1 in both the function signature and the YAML configs in `composer` (all public facing). 
2. we run our release testing sweeps with "auto"  + add some additional YAML configs that stress test this feature
3. we make `grad_accum=auto` the default in our internal `mcli` tools so internally we are testing it.

We let it bake for a month or two until we set the default with a major increment. We will likely get some additional feedback on the design in this burn-in period. (e.g. if i _know_ grad accum needs to be around 50, why wait all this latency while the script tried to reset itself?)

Reply:
Got it, thanks for clearing up the process. Will adjust configs accordingly